frontend/src/components/Dropdown/index.jsx

```diff
@@ -0,0 +1,148 @@
+import styled from '@emotion/styled';
+import React, { useState } from 'react';
+import DropdownItem from '../DropdownItem'
+
+function Dropdown( {suggestions, addSuggestion} ) {
+  
+  const [activeSuggestionIndex, setActiveSuggestionIndex] = useState(-1);
+  const [filteredSuggestions, setfilteredSuggestions] = useState([]);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [userInput, setUserInput] = useState("");
+
+  // TODO: 더 나은 필터 방식 적용하기
+  const onInputChange = e => {
+    if (suggestions.length) {
+      const filtered = suggestions.filter(suggestion => (
+        suggestion.toLowerCase().indexOf(userInput.toLowerCase()) > -1
+      ));
+      setfilteredSuggestions(filtered);
```

Comment:
```suggestion
      setfilteredSuggestions(prev => {
        if (prev.length === 0) {
          return prev;
        }
      
        return prev.filter(s => s.toLowerCase().indexOf(e.target.value.toLowerCase()) > -1);
      });
```

요렇게 하면 onInputChange를 useCallback 으로 바꾸어도 deps 가 없게 잡힐 겁니다!

Reply:
피드백 감사합니다!
필터되는 대상이 filteredSuggestions이 아닌 그냥 suggestions(=AllTasks를 받음)이기 때문에 
prev를 대상으로 filter 함수를 사용할 경우, autocomplete가 제대로 실행되지 않을 것 같네요.
대신, filter 대상을 그대로 suggestions로 납두되,
useCallback을 사용하는 방식으로 주신 피드백을 반영하였습니다!